Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/compression/ZLIBCompression.kt

```diff
@@ -1,24 +1,34 @@
 package de.rki.coronawarnapp.util.compression
 
+import okio.Buffer
 import okio.ByteString
-import okio.ByteString.Companion.toByteString
-import timber.log.Timber
-import java.util.zip.InflaterInputStream
+import okio.inflate
+import java.util.zip.Inflater
 import javax.inject.Inject
 
 class ZLIBCompression @Inject constructor() {
-    fun decompress(input: ByteString): ByteString = if (
-        input.size >= 2 &&
-        input[0] == 0x78.toByte() &&
-        input[1] in listOf(0x01.toByte(), 0x5E.toByte(), 0x9C.toByte(), 0xDA.toByte())
-    ) {
-        try {
-            input.toByteArray().inputStream().use { InflaterInputStream(it).readBytes().toByteString() }
-        } catch (e: Throwable) {
-            Timber.e(e)
-            throw InvalidInputException("Zlib decompression failed.")
+    fun decompress(input: ByteString, sizeLimit: Long = -1L): ByteString = try {
+        val inflaterSource = input.let {
```

Comment:
Where are the zlib magic numbers?

Reply:
A zlib(inflate) header check is part of `java.util.zip.Inflater`. I did not see a specific requirement for extra header checks in the tech spec, so I thought the default checks are sufficient.

Added a test for this:
https://github.com/corona-warn-app/cwa-app-android/pull/3144/commits/44a4ce5b7b4e95c7a980218ddfbd045e8031d79a#diff-b710f361f25e20d66b7b634de24f4e3017fd9e3f52fede9f4b3fa8a97fb22c21R18-R25

If the header doesn't match we get:

```java
de.rki.coronawarnapp.util.compression.InvalidInputException: ZLIB decompression failed.

	at de.rki.coronawarnapp.util.compression.ZLIBCompression.decompress(ZLIBCompression.kt:29)
	at de.rki.coronawarnapp.util.compression.ZLIBCompression.decompress$default(ZLIBCompression.kt:11)
	at de.rki.coronawarnapp.util.compression.ZLIBCompressionTest.basic decompression(ZLIBCompressionTest.kt:15)
(....)
Caused by: java.io.IOException: java.util.zip.DataFormatException: incorrect header check
	at okio.InflaterSource.readOrInflate(InflaterSource.kt:99)
	at de.rki.coronawarnapp.util.compression.ZLIBCompression.decompress(ZLIBCompression.kt:22)
	... 67 more
Caused by: java.util.zip.DataFormatException: incorrect header check
	at java.base/java.util.zip.Inflater.inflateBytesBytes(Native Method)
	at java.base/java.util.zip.Inflater.inflate(Inflater.java:385)
	at okio.InflaterSource.readOrInflate(InflaterSource.kt:79)
	... 68 more
```

Reply:
OK. It was in the reference implementation, that's why I included it.